cmd/virt-apiservice-register/README.md

```diff
@@ -0,0 +1,193 @@
+  version exists. If it doesn't the tool creates it. The name of the
+  secret can be changed with the `-secret-name` command line option.
+
+* Checks if the secret contains a `ca.crt` key, and if that key contains
+  the CA certificate tht Kubernetes uses to sign certificates for API
+  services. If the secret doesn't contain that key, then the tool finds
+  the CA certificate and updates the secret.
+
+* Checks if the secret contains a `tls.key` key, and if that key
+  contains a private key. If the secret doesn't contain that key, then
+  the tool generates a new one and updates the secret.
+
+* Checks if the secret contains a `tls.crt` key, and if that key
+  contains a certificate. If the secret doesn`t contain that
+  certificate, then the tool checks if there is a certificate signing
+  request for the private key, creates it if needed, and waits till it
+  is approved. When the certificate signing request is approved the tool
+  updates the secret.
+
+* Checks if the service that handles the traffic to the API server
+  exists. If it doesn't exist then the tool creates it. By default the
+  name of this service is `apiservice` followed by the API group and
+  version, but it can be changed using the optional `-service-name`
+  command line option.
+
+* Checks if the the API service is registered. If it isn't registered
+  then the tool registers it. The name used to register the API service
+  is the API version followed by a dot and the API group. It can't be
+  changed, as this is required by Kubernetes.
+
+For example, to create register the service for API group `myapi.io` and
+version `v1alpha` the command could be like this:
+
+```shell
+virt-apiservice-register \
+-api-group=myapi.io \
+-api-version=v1alpha1 \
+-secret-name=mysecret \
+-service-name=myservice \
+-target-selector=app=myapiservice
+```
+
+It is important to make sure that the target, the pod where the actual
+API service is running, has labels that matches the selector given in the
+'-target-selector' option.
+
+## How to use it
+
+The tool is intended for use in an initialization container inside the
+pod of the API server itself, so that it is always executed before the
+API server container.
+
+The tool can create the secret if it doesn't exist, but that doesn't
+work if the secret is going to be mounted by the pod, as Kubernetes will
+not start the pod till the secret exists. In this case the secret needs
+to be created manually, ideally as part of the same manifests that
+creates the pod where the tool will be executed. For example:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+---
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: myserver
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: myserver
+    spec:
+      serviceAccountName: myaccount
+      volumes:
+      - name: pkifiles
+        secret:
+          secretName: mysecret
+      initContainers:
```

Comment:
I we should remove talking about initContainers. A vague "this tool must run before the api server can initialize" is enough for now. We're still sorting through how this will be deployed.